src/sas/sascalc/pr/p_invertor.py

```diff
@@ -0,0 +1,663 @@
+    y = np.empty(0, dtype=np.float)
+    #dI(q) data
+    err = np.empty(0, dtype=np.float)
+    #Number of q points
+    npoints = 0
+    #Number of I(q) points
+    ny = 0
+    #Number of dI(q) points
+    nerr = 0
+    #Slit height in units of q [A-1]
+    slit_height = 0.0
+    #Slit width in units of q [A-1]
+    slit_width = 0.0
+
+    def __init__(self):
+        #Maximum distance between any two points in the system
+        self.set_dmax(180)
+        #Minimum q to include in inversion
+        self.set_qmin(-1.0)
+        #Maximum q to include in inversion
+        self.set_qmax(-1.0)
+        #Flag for whether or not to evaluate a constant background
+        #while inverting
+        self.set_est_bck(0)
+        self.set_alpha(0.0)
+
+    def residuals(self, pars):
+        """
+        Function to call to evaluate the residuals for P(r) inversion.
+
+        :param pars: input parameters.
+        :return: residuals - list of residuals.
+        """
+        pars = np.float64(pars)
+
+        residuals = []
+        nslice = 25
+        regterm = calc.reg_term(pars, self.d_max, nslice)
+        resid = (self.y[0:self.npoints] - calc.iq(pars, self.d_max, self.x))/self.err
+
+        return list(resid**2 + self.alpha*regterm)
+
+    def pr_residuals(self, pars):
+        """
+        Function to call to evaluate the residuals for P(r) minimization (for testing purposes).
+
+        :param pars: input parameters.
+        :return: residuals - list of residuals.
+        """
+        pars = np.float64(pars)
+
+        residuals = []
+        nslice = 25
+        regterm = calc.reg_term(pars, self.d_max, nslice)
+        resid = (self.y[0:npoints] - calc.pr(pars, self.d_max, self.x))/self.err
+        return list(resid**2 + self.alpha * regterm)
+
+    def set_x(self, data):
+        """
+        Function to set the x data.
+
+        :param data: Array of doubles to set x to.
+        :return: npoints - Number of entries found, the size of x.
+        """
+        data = np.float64(data)
+        ndata = len(data)
+        self.__dict__['x'] = np.zeros(ndata)
+
+        self.__dict__['x'] = np.array(data, 'd')
+
+        self.__dict__['npoints'] = int(ndata)
+        return self.npoints
+
+    def get_x(self, data):
+        """
+        Function to get the x data.
+
+        :param data: Array to place x into
+        :return: npoints - Number of entries found.
+        """
+        data = np.float64(data)
+        ndata = len(data)
+
+        if ndata < self.npoints:
+            logger.error("Pinvertor.get_x: input array too short for data.")
+            return None
+
+        data[:] = self.x
+
+        return self.npoints
+
+    def set_y(self, data):
+        """
+        Function to set the y data.
+
+        :param data: Array of doubles to set y to.
+        :return: ny - Number of entries found.
+        """
+        data = np.float64(data)
+        ndata = len(data)
+        self.__dict__['y'] = np.zeros(ndata)
+
```

Comment:
`y` gets overwritten on next line; don't need to initialize it.

Reply:
Done.